ICSharpCode.Decompiler/IL/Transforms/ILInlining.cs

```diff
@@ -30,28 +30,36 @@ public class ILInlining : IILTransform, IBlockTransform, IStatementTransform
 	{
 		public void Run(ILFunction function, ILTransformContext context)
 		{
+			int? ctorCallStart = null;
 			foreach (var block in function.Descendants.OfType<Block>()) {
-				InlineAllInBlock(block, context);
+				InlineAllInBlock(function, block, context, ref ctorCallStart);
 			}
 			function.Variables.RemoveDead();
 		}
 
 		public void Run(Block block, BlockTransformContext context)
 		{
-			InlineAllInBlock(block, context);
+			InlineAllInBlock(context.Function, block, context);
 		}
 
 		public void Run(Block block, int pos, StatementTransformContext context)
 		{
 			InlineOneIfPossible(block, pos, aggressive: IsCatchWhenBlock(block), context: context);
 		}
 
-		public static bool InlineAllInBlock(Block block, ILTransformContext context)
+		public static bool InlineAllInBlock(ILFunction function, Block block, ILTransformContext context)
+		{
+			int? ctorCallStart = null;
+			return InlineAllInBlock(function, block, context, ref ctorCallStart);
+		}
+
+		static bool InlineAllInBlock(ILFunction function, Block block, ILTransformContext context, ref int? ctorCallStart)
 		{
 			bool modified = false;
-			int i = 0;
-			while (i < block.Instructions.Count) {
-				if (InlineOneIfPossible(block, i, aggressive: IsCatchWhenBlock(block) || IsInConstructorInitializer(i, block), context: context)) {
+			var instructions = block.Instructions;
+			for (int i = 0; i < instructions.Count;) {
```

Comment:
Why did you replace the while loop with a for loop? A for loop without increment is a bit confusing at first sight.

Reply:
It's more compact and the loop variable is better scoped, but it's a matter of taste. Should I change it back?

Reply:
Well... I guess we can leave it like it is, I don't think it's worth the time to change it back.